utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala

```diff
@@ -0,0 +1,278 @@
+package pl.touk.nussknacker.engine.util.functions
+
+import cats.data.ValidatedNel
+import cats.implicits.catsSyntaxValidatedId
+import pl.touk.nussknacker.engine.api.generics.{GenericFunctionTypingError, GenericType, TypingFunction}
+import pl.touk.nussknacker.engine.api.typed.typing
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedClass, TypedObjectTypingResult, Unknown}
+import pl.touk.nussknacker.engine.api.{Documentation, HideToString, ParamName}
+
+import java.util.{Collections, Objects}
+import scala.collection.immutable.ListMap
+import scala.jdk.CollectionConverters._
+import scala.language.higherKinds
+import scala.reflect.ClassTag
+
+object collection extends HideToString {
+
+  @Documentation(description = "Concatenates two lists")
+  @GenericType(typingFunction = classOf[ListAdditionTyping])
+  def concat[T](@ParamName("list1") list1: java.util.List[T], @ParamName("list2") list2: java.util.List[T]): java.util.List[T] =
+    (list1.asScala.toList ++ list2.asScala).asJava
+
+  @Documentation(description = "Merges two maps. Values in the first map will be overwritten with values from the second map if keys are the same")
+  @GenericType(typingFunction = classOf[MapMergeTyping])
+  def merge[K, V](@ParamName("map1") map1: java.util.Map[K, V], @ParamName("map2") map2: java.util.Map[K, V]): java.util.Map[K, V] = {
+    val merged = new java.util.LinkedHashMap[K, V](map1)
+    merged.putAll(map2)
+    merged
+  }
+
+  @Documentation(description = "Returns the smallest element (elements must be comparable)")
+  @GenericType(typingFunction = classOf[ListElementTyping])
+  def min[T <: Comparable[T]](@ParamName("list") list: java.util.Collection[T]): T = Collections.min[T](list)
+
+  @Documentation(description = "Returns the largest element (elements must be comparable)")
+  @GenericType(typingFunction = classOf[ListElementTyping])
+  def max[T <: Comparable[T]](@ParamName("list") list: java.util.Collection[T]): T = Collections.max[T](list)
+
+  @Documentation(description = "Returns a slice of the list starting with start index (inclusive) and ending at stop index (exclusive)")
+  @GenericType(typingFunction = classOf[ListTyping])
+  def slice[T](@ParamName("list") list: java.util.Collection[T], @ParamName("start") start: Int, @ParamName("stop") stop: Int): java.util.List[T] =
+    list.asScala.slice(start, stop).toList.asJava
+
+  private val numberPriorities = Map(
+    classOf[java.lang.Short] -> 1,
+    classOf[java.lang.Byte] -> 2,
+    classOf[java.lang.Integer] -> 3,
+    classOf[java.lang.Long] -> 4,
+    classOf[java.math.BigInteger] -> 5,
+    classOf[java.lang.Float] -> 6,
+    classOf[java.lang.Double] -> 7,
+    classOf[java.math.BigDecimal] -> 8,
+  ).asInstanceOf[Map[Class[_], Int]]
+
+  private def getNumberPriority(number: Number): Int = numberPriorities
```

Comment:
`getNumberTypePriority` (and `numberPriorities`)?

Reply:
Makes, sense, but it was removed anyway :)